lib/os/cbprintf_complete.c

```diff
@@ -1815,7 +1815,7 @@ int z_cbvprintf_impl(cbprintf_cb out, void *ctx, const char *fp,
 
 			OUTS(cp, bpe);
 		} else {
-			if (conv->altform_0c | conv->altform_0) {
+			if ((conv->altform_0c | conv->altform_0) != 0) {
```

Comment:
would this be better as if (conv->altform_0c !=0 || conv->altform_0 != 0) ? (I haven't looked at the surrounding code)

Reply:
I think it would be more readable if we change it. But than i would recommend also doing if ((conv->altform_0c != false) || (conv->altform_0 != false )) because altform and altform_0c are of boolean type. I can live with both solutions what would be your recomendation?

Reply:
Sorry, I hadn't looked at the underlying types of the two values. As they're both boolean, then the correct code
would just be `if (conv->altform_0c || conv->altform_0)` -- switch from the bitwise to the boolean `or` operator.

Reply:
Thank you, I will adapt the commit